pkg/router/router_test.go

```diff
@@ -203,6 +211,18 @@ func TestConfigTemplateExecution(t *testing.T) {
 				tlsTermination: routev1.TLSTerminationEdge,
 			},
 		},
+		"case insesitive, with white spaces HSTS header": {
+			mustCreate{
+				name: "h",
+				host: "hstsexample.com",
+				path: "",
+				time: start,
+				annotations: map[string]string{
+					"haproxy.router.openshift.io/hsts_header": hstsHeaderValue,
+				},
+				tlsTermination: routev1.TLSTerminationEdge,
+			},
+		},
```

Comment:
One more test that would fail is good.

Reply:
I added some more patterns and not matching patterns too.
Now it should easier to understand what is expected and what is not.

It also helped me to find a problem in the config template, I used double quotes for `Strict-Transport-Security` but `max-age` can have double quotes too. This may be badly interpreted by HAProxy.